torch/csrc/autograd/engine.cpp

```diff
@@ -352,6 +340,26 @@ auto Engine::thread_main(GraphTask *graph_task) -> void {
       }
     }
   }
+
+  if (graph_task) {
+    graph_task->not_done_.notify_all();
+  }
+}
+
+void Engine::reentrant_thread_init() {
```

Comment:
So this function gets called only for the "extra" threads that we launch if the "main" ones get blocked by reentrant calls?

Reply:
that is correct.